dds/DCPS/Barrier.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_BARRIER_H
+#define OPENDDS_DCPS_BARRIER_H
+
+#include "dcps_export.h"
+
+#include "ConditionVariable.h"
+#include "ThreadStatusManager.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+/**
+ * Barrier provides thread-coordination across an expected number of threads
+ *
+ * Threads which call Barrier::wait() will all wait until the expected number
+ * of threads have called Barrier::wait(), at which point they will all be woken
+ * up and the next "phase" of waiting will begin (again, being released when the
+ * expected number of waiting threads is reached
+ */
+class OpenDDS_Dcps_Export Barrier {
```

Comment:
Why not use ACE_Barrier?

Reply:
I didn't even think to check if ACE had a barrier class. It doesn't look too heavy-weight and the API is mostly the same, so I'll give it a shot, thanks.